services/provider/server/server.go

```diff
@@ -107,23 +111,47 @@ func NewOCSProviderServer(ctx context.Context, namespace string) (*OCSProviderSe
 		return nil, fmt.Errorf("failed to create new scheme. %v", err)
 	}
 
-	client, err := util.NewK8sClient(scheme)
+	config, err := config.GetConfig()
 	if err != nil {
-		return nil, fmt.Errorf("failed to create new client. %v", err)
+		return nil, err
 	}
 
-	consumerManager, err := newConsumerManager(ctx, client, namespace)
+	mgr, err := manager.New(config, manager.Options{
+		Scheme: scheme,
+		Cache: cache.Options{
+			DefaultNamespaces: map[string]cache.Config{
+				namespace: {},
+			},
+		},
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	consumerManager, err := newConsumerManager(ctx, mgr, namespace)
```

Comment:
same, rather than sending manager, could do `mgr.GetClient()`

Reply:
I am creating the informer to populate the nameByUID map in newConsumerManager, and it requires a cache. I am using the cache from mgr instead of creating a new one, hence passing the mgr instead of the client

Reply:
ok, missed it. thanks.